packages/driver/cypress/e2e/commands/assertions.cy.js

```diff
@@ -1408,6 +1408,20 @@ describe('src/cy/commands/assertions', () => {
 
         cy.get('#does-not-exist').should('not.exist')
       })
+
+      context('with a jQuery selector', () => {
+        it('can be chained', () => {
+          cy.get('#foo').should('exist').and('exist')
+        })
+      })
+
+      context('with an HTML element', () => {
+        it('can be chained', () => {
+          cy.document().then((doc) => {
+            cy.wrap(doc.getElementById('foo')).should('exist').and('exist')
```

Comment:
What I find strange is the differing behavior that wrap gives you vs using a `.then`. For example:

```js
cy.document().then((doc) => doc.getElementById('foo'))
  .should('exist').and('exist')
```

Works just fine. Also, using a `.then` on the wrap returns a jquery element e.g.

```js
cy.document().then((doc) => {
  return cy.wrap(doc.getElementById('foo')).then(($el) => {
    console.log({ $el }) // {$el: jQuery.fn.init(1)}
  })
})
```

But for the case you have pointed out, the subject is the raw html element and not a jquery-able (if that's a word). I would expect the subject the `should` assertion receives would be the same as the result of the `.then` which makes me think the bug is somewhere else then what you have modified.

@BlueWinds could I get your thoughts on this?

Reply:
I have a PR in progress which basically rewrites how assertions work entirely - I'm wondering if this bug still reproduces in https://github.com/cypress-io/cypress/pull/25296 .

I'll take a closer look into this next week, but it doesn't surprise me that .wrap() is inconsistent in behavior vs. other commands, it's a weird one.

Reply:
@BlueWinds Did you have a chance to look at this issue again this week to see if it is fixed in the PR you referenced?

Reply:
I had forgotten, thank you for the reminder!

It does indeed appear to be fixed in https://github.com/cypress-io/cypress/pull/25296 - both tests added here pass on that branch. I've added them to that PR in this commit: https://github.com/cypress-io/cypress/pull/25296/commits/d6ad5abecf31a85b2175400994f4c102b27f8d41

That PR should hopefully be merged into the 13.x branch sometime next week.

Reply:
@thoughtbot-summer - If you're ok with waiting for 13.x, feel free to close this PR; otherwise, we can continue to review / pursue the fix you have here.

Reply:
Sounds good to me; thanks for your work! :heart: